diff --git a/node_modules/react-native-track-player/android/src/main/java/com/doublesymmetry/trackplayer/module/MusicModule.kt b/node_modules/react-native-track-player/android/src/main/java/com/doublesymmetry/trackplayer/module/MusicModule.kt
index b2409a0..48736ef 100644
--- a/node_modules/react-native-track-player/android/src/main/java/com/doublesymmetry/trackplayer/module/MusicModule.kt
+++ b/node_modules/react-native-track-player/android/src/main/java/com/doublesymmetry/trackplayer/module/MusicModule.kt
@@ -542,10 +542,15 @@ class MusicModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaM
 
     @ReactMethod
     fun getTrack(index: Int, callback: Promise) = scope.launch {
+        val bundle = musicService.tracks[index].originalItem
         if (verifyServiceBoundOrReject(callback)) return@launch
 
         if (index >= 0 && index < musicService.tracks.size) {
-            callback.resolve(Arguments.fromBundle(musicService.tracks[index].originalItem))
+            callback.resolve(
+                bundle?.let {
+                    Arguments.fromBundle(it)
+                }
+            )
         } else {
             callback.resolve(null)
         }
@@ -581,12 +586,13 @@ class MusicModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaM
 
     @ReactMethod
     fun getActiveTrack(callback: Promise) = scope.launch {
+        val bundle = musicService.tracks[musicService.getCurrentTrackIndex()].originalItem
+
         if (verifyServiceBoundOrReject(callback)) return@launch
         callback.resolve(
-            if (musicService.tracks.isEmpty()) null
-            else Arguments.fromBundle(
-                musicService.tracks[musicService.getCurrentTrackIndex()].originalItem
-            )
+            bundle?.let {
+                Arguments.fromBundle(it)
+            }
         )
     }
 
